tests/integration/conditions.py

```diff
@@ -0,0 +1,207 @@
+
+
+class StudyCond(Condition):
+    """
+    An abstract condition that is aware of a study's name and output path.
+    """
+
+    def __init__(self, study_name, output_path):
+        """
+        :param `study_name`: the name of a study
+        :param `output_path`: the $(OUTPUT_PATH) of a study
+        """
+        self.study_name = study_name
+        self.output_path = output_path
+        self.dirpath_glob = f"{self.output_path}/{self.study_name}" f"_[0-9]*-[0-9]*"
+
+    def glob(self, glob_string):
+        candidates = glob(glob_string)
+        if isinstance(candidates, list):
+            return sorted(candidates)[-1]
+        return candidates
+
+
+class StepFileExistsCond(StudyCond):
+    """
+    A StudyCond that checks for a particular file's existence.
+    """
+
+    def __init__(self, step, filename, study_name, output_path, params=False):
+        """
+        :param `step`: the name of a step
+        :param `filename`: name of file to search for in step's workspace directory
+        :param `study_name`: the name of a study
+        :param `output_path`: the $(OUTPUT_PATH) of a study
+        """
+        super().__init__(study_name, output_path)
+        self.step = step
+        self.filename = filename
+        self.params = params
+
+    def file_exists(self):
+        param_glob = ""
+        if self.params:
+            param_glob = "*/"
+        glob_string = f"{self.dirpath_glob}/{self.step}/{param_glob}{self.filename}"
+        try:
+            filename = self.glob(glob_string)
+        except IndexError:
+            return False
+        return os.path.isfile(filename)
+
+    @property
+    def passes(self):
+        return self.file_exists()
+
+
+class StepFileContainsCond(StudyCond):
+    """
+    A StudyCond that checks that a particular file contains a regex.
+    """
+
+    def __init__(self, step, filename, study_name, output_path, regex):
+        """
+        :param `step`: the name of a step
+        :param `filename`: name of file to search for in step's workspace directory
+        :param `study_name`: the name of a study
+        :param `output_path`: the $(OUTPUT_PATH) of a study
+        """
+        super().__init__(study_name, output_path)
+        self.step = step
+        self.filename = filename
+        self.regex = regex
+
+    def contains(self):
+        glob_string = f"{self.dirpath_glob}/{self.step}/{self.filename}"
+        try:
+            filename = self.glob(glob_string)
+            with open(filename, "r") as textfile:
+                filetext = textfile.read()
+            return self.is_within(filetext)
+        except Exception:
+            return False
+
+    def is_within(self, text):
+        """
+        :param `text`: text in which to search for a regex match
+        """
+        return search(self.regex, text) is not None
+
+    @property
+    def passes(self):
+        return self.contains()
+
+
+class ProvenanceCond(RegexCond):
```

Comment:
I think we can improve the name of this object in a future change (eg ProvenanceHasCond or ProvenanceContainsCond).

We could also consider dropping "Cond" from the names and making them more descriptive. 

Right now I understand this is just a re-org PR so don't do any of this here, but let's considering knocking some of this technical debt later.

Reply:
Okay, I can make an issue for this.